x-pack/legacy/plugins/maps/common/data_request_descriptor_types.d.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+/* eslint-disable @typescript-eslint/consistent-type-definitions */
+
+// Global map state passed to every layer.
+export type MapFilters = {
+  buffer: unknown;
+  extent: unknown;
+  filters: unknown[];
+  query: unknown;
+  refreshTimerLastTriggeredAt: string;
+  timeFilters: unknown;
+  zoom: number;
+};
+
+export type VectorLayerRequestMeta = MapFilters & {
+  applyGlobalQuery: boolean;
+  fieldNames: string[];
+  geogridPrecision: number;
+  sourceQuery: unknown;
+  sourceMeta: unknown;
+};
+
+export type ESSearchSourceResponseMeta = {
+  areResultsTrimmed?: boolean;
+  sourceType?: string;
+
+  // top hits meta
+  areEntitiesTrimmed?: boolean;
+  entityCount?: number;
+  totalEntities?: number;
+};
+
+export type DataMeta = Partial<VectorLayerRequestMeta> & Partial<ESSearchSourceResponseMeta>;
```

Comment:
would you mind adding a comment for this? it's an obscure TS-feature. Making this all partial is really here because we justify these objects downstream in constructor.